signalfx-java/src/main/java/com/signalfx/signalflow/WebSocketTransport.java

```diff
@@ -0,0 +1,592 @@
+                    }
+                } else {
+                    log.warn("Unsupported binary message type {}", messageType);
+                }
+            } catch (Exception ex) {
+                log.error("failed to construct transport data message", ex);
+            }
+        }
+
+        public int getVersion() {
+            return (int) version;
+        }
+
+        public String getChannelName() {
+            return channelName;
+        }
+
+        public int getMessageType() {
+            return (int) messageType;
+        }
+
+        public List<Map<String, Object>> getData() {
+            return data;
+        }
+
+        public long getLogicalTimestampMs() {
+            return logicalTimestampMs;
+        }
+    }
+
+    /**
+     * WebSocket Transport Connection
+     */
+    protected static class TransportConnection
+            implements WebSocket.OnTextMessage, WebSocket.OnBinaryMessage {
+
+        protected static final Logger log = LoggerFactory.getLogger(TransportConnection.class);
+        protected String token;
+        protected SignalFlowException error;
+        protected WebSocket.Connection connection;
+        protected Map<String, TransportChannel> channels = Collections
+                .synchronizedMap(new HashMap<String, TransportChannel>());
+        protected static ObjectMapper objectMapper = new ObjectMapper();
+        static {
+            objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        }
+
+        public TransportConnection(String token) {
+            this.token = token;
+        }
+
+        @Override
+        public void onClose(int code, String reason) {
+            if (log.isDebugEnabled()) {
+                log.debug("websocket connection closed - code: " + code + " reason: " + reason);
+            }
+
+            if (code != 1000) {
+                this.error = new SignalFlowException(code, reason);
+                log.info("Lost WebSocket connection with {} ({}).", connection, code);
+
+                SignalFlowExceptionStreamMessage errorMessage = new SignalFlowExceptionStreamMessage(
+                        this.error);
+                for (TransportChannel channel : this.channels.values()) {
+                    channel.offer(errorMessage);
+                }
+            }
+
+            this.channels.clear();
+            this.connection = null;
+        }
+
+        @Override
+        public void onOpen(Connection connection) {
+            if (log.isDebugEnabled()) {
+                log.debug("open connection: " + connection);
+            }
+            this.connection = connection;
+
+            Map<String, String> authRequest = new HashMap<String, String>();
+            authRequest.put("type", "authenticate");
+            authRequest.put("token", this.token);
+
+            sendMessage(authRequest);
+        }
+
+        @Override
+        public void onMessage(byte[] data, int offset, int length) {
+
```

Comment:
done but my preference is to give a line under method signature for better viewing.
you obviously have a different opinion.